oauth2_http/java/com/google/auth/oauth2/UserAuthorizer.java

```diff
@@ -291,9 +309,11 @@ public UserCredentials getCredentialsFromCode(
     GenericData tokenData = new GenericData();
     tokenData.put("code", code);
     tokenData.put("client_id", clientId.getClientId());
-    tokenData.put("client_secret", clientId.getClientSecret());
     tokenData.put("redirect_uri", resolvedCallbackUri);
     tokenData.put("grant_type", "authorization_code");
+    if (this.clientAuthenticationType == ClientAuthenticationType.CLIENT_SECRET_POST) {
```

Comment:
Can you group this in an if/else with the if statement below?

Reply:
I'm afraid not here. In the Java code practice we are having a token request content built here and from this content to generate a request build. After the request build, we adding proper basic auth (if chosen) to the request.